app/src/main/java/com/ac03/covid/CovidApplication.kt

```diff
@@ -0,0 +1,39 @@
+package com.ac03.covid
+
+import android.app.Application
+import com.facebook.flipper.android.AndroidFlipperClient
+import com.facebook.flipper.android.utils.FlipperUtils
+import com.facebook.flipper.plugins.inspector.DescriptorMapping
+import com.facebook.flipper.plugins.inspector.InspectorFlipperPlugin
+import com.facebook.flipper.plugins.network.NetworkFlipperPlugin
+import com.facebook.soloader.SoLoader
+
+class CovidApplication : Application() {
+    val flipperNetworkPlugin by lazy { NetworkFlipperPlugin() }
+
+    companion object {
+        private var INSTANCE: CovidApplication? = null
```

Comment:
We only have one instance, and we need a reference because I am adding an interceptor for HTTP requests in CovidServiceFactory.

Reply:
Sorry, you are right! In this case only we are using a new reference to the instance of CovidApplication. Otherwise, I would avoid that it can be nullable and I would define it as _lateinit_
` private lateinit var INSTANCE: CovidApplication`

Reply:
updated :)